kuka_rox_hardware_interface/src/robot_manager_node.cpp

```diff
@@ -46,64 +48,60 @@ RobotManagerNode::RobotManagerNode()
     "robot_manager/is_configured",
     is_configured_qos);
 
-  control_mode_map_.emplace(
-    std::make_pair(
-      ExternalControlMode::POSITION_CONTROL,
-      std::vector<std::string>(STANDARD_MODE_IF_SIZE)));
-  control_mode_map_.emplace(
-    std::make_pair(
-      ExternalControlMode::JOINT_IMPEDANCE_CONTROL,
-      std::vector<std::string>(IMPEDANCE_MODE_IF_SIZE)));
-  control_mode_map_.emplace(
-    std::make_pair(
-      ExternalControlMode::TORQUE_CONTROL,
-      std::vector<std::string>(STANDARD_MODE_IF_SIZE)));
-
-  // TODO(Svastits): change to dynamic parameter after control mode changes are supported
-  this->registerStaticParameter<int>(
+  control_mode_pub_ = this->create_publisher<std_msgs::msg::UInt32>(
+    "control_mode", rclcpp::SystemDefaultsQoS()
+  );
+
+  control_mode_change_finished_ = false;
+
+  // Register parameters
+  this->registerParameter<int>(
     "control_mode", static_cast<int>(ExternalControlMode::POSITION_CONTROL),
-    kroshu_ros2_core::ParameterSetAccessRights{true, false,
-      false, false, false}, [this](int control_mode) {
-      return this->onControlModeChangeRequest(control_mode);
+    kroshu_ros2_core::ParameterSetAccessRights{true, true,
+      true, false, false}, [this](int control_mode) {
+      // TODO (komaromi):
+      if (control_mode != 2 && control_mode != 4 && control_mode != 6) {
```

Comment:
The function itself does not have condition for not implemented control modes, because it was created in mind that it can work with every control mode we defined

Also I forgot to write the todo: It has to be removed once control modes are supported

Reply:
The reason I'd handle it inside the function is that you won't have to guard against this outside the function every single time. This is probably not a big deal, but prevents accidental mistakes. It's also not a good practice to introduce contracts whenever you wish to use a function, since the developer / user might not know about those without looking through the source code in detail.